### PR TITLE
[WIP] Series list

### PR DESCRIPTION
**Pull request by liiight**
### Motivation for changes:

Add an internal way to hold a series list
### Detailed changes:
- Added managed list `series_list` plugin
- Created `SeriesList` DB
- Created `list_id` column in `Series` DB and added a foreign key to `SeriesList` DB
- Created a migration script
### Addressed issues:
- Fixes # .
### Config usage if relevant (new plugin or updated schema):

``` yaml
  test_series_list:
    mock:
      - {series_name: "lucifer", title: "lucifer"}
    list_add:
      - series_list: series_list
    accept_all: yes
```
#### To Do:
- [ ] Create default list if `series` is defined in config.
- [ ] API / CLI 
- [ ] Tests

### CODE REVIEW

File: flexget/plugins/filter/series.py [hunk 1]
```diff
@@ -28,7 +28,8 @@
                                             create_index)
 from flexget.utils.tools import merge_dict_from_to, parse_timedelta, parse_episode_identifier
 
-SCHEMA_VER = 13
+SCHEMA_VER = 14
+DEFAULT_SERIES_LIST_NAME = 'series_list'
```
Comment (paranoidi) [inline flexget/plugins/filter/series.py]: Default name should be "series" like with movies it's "movies"

(in hindsight I would like it to have been default in all places)

File: flexget/plugins/list/series_list.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import logging
+from collections import MutableSet
+from datetime import datetime
+
+from flexget import plugin
+from flexget.db_schema import versioned_base
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.manager import Session
+from flexget.plugins.filter.series import Series
```
Comment (paranoidi) [inline flexget/plugins/list/series_list.py]: Does not handle missing dependencies

File: flexget/plugins/filter/series.py [hunk 3]
```diff
@@ -154,6 +155,12 @@ def upgrade(ver, session):
         series_table = table_schema('series', session)
         session.execute(update(series_table, series_table.c.identified_by == None, {'identified_by': 'auto'}))
         ver = 13
+    if ver == 13:
+        # Create default series list
+        series_list_table = table_schema('series_list_lists', session)
+        result = session.execute(series_list_table.insert().values(name=DEFAULT_SERIES_LIST_NAME, added=datetime.now()))
+        table_add_column('series', 'list_id', Integer, session, result.inserted_primary_key[0])
```
Comment (paranoidi) [inline flexget/plugins/filter/series.py]: Not sure about relationships across plugins .. doesn't seem like a good idea. Series list would be a core part of series plugin so it should probably be in the ever so large series plugin ...

Comment (liiight) [inline flexget/plugins/filter/series.py]: :expressionless:

File: flexget/plugins/filter/series.py (hunk 1 above)
Comment (paranoidi) [inline flexget/plugins/filter/series.py]: also configuring

```
series_list: series_list
```

seems retarded

File: flexget/plugins/filter/series.py (hunk 3 above)
Comment (stevezau) [inline flexget/plugins/filter/series.py]: @paranoidi does it not feel weird that all other lists are in plugins/list but this is in filter/series?

**Comment by stevezau** [comment]
Closing this, i will reopen a new PR

**Pull request closed without merging**